src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1172,6 +1179,247 @@ void SolverInterfaceImpl::writeScalarData(
+                data.getName(), data.getName());
+
+  PRECICE_CHECK(data.getSpacialDimensions() == _dimensions,
+                "The spatial dimensions of the data object and of the mesh don't match. Please make sure to initialize {} space dimensions for the scalar data type \"{}\".",
+                _dimensions, context.getDataName());
+
+  PRECICE_VALIDATE_DATA(value, _dimensions * _dimensions);
+
+  auto &     gradientValuesInternal = data.gradientValues();
+  const auto vertexCount            = gradientValuesInternal.cols() / data.getDimensions();
+
+  // Check if the index is valid
+  PRECICE_CHECK(0 <= valueIndex && valueIndex < vertexCount,
+                "Cannot write gradient data \"{}\" to invalid Vertex ID ({}). Please make sure you only use the results from calls to setMeshVertex/Vertices().",
+                data.getName(), valueIndex)
+
+  for (int dimSpace = 0; dimSpace < _dimensions; dimSpace++) {
+    for (int dimData = 0; dimData < _dimensions; dimData++) {
+
+      const int offsetInternal = valueIndex * _dimensions;
+
+      if (rowMajor) {
+        // Values are entered derived in spatial dimensions first : gradient matrix read rowwise
+        const int offset                                           = dimData * _dimensions;
+        gradientValuesInternal(dimSpace, offsetInternal + dimData) = value[offset + dimSpace];
+      } else {
+        // Values are entered derived components first : gradient matrix read columnwise
+        const int offset                                           = dimSpace * _dimensions;
+        gradientValuesInternal(dimSpace, offsetInternal + dimData) = value[offset + dimData];
+      }
+
+      //TODO: Add asserts
+    }
+  }
+}
+
+void SolverInterfaceImpl::writeBlockVectorGradientData(
+    int           dataID,
+    int           size,
+    const int *   valueIndices,
+    const double *values,
+    bool          rowMajor)
+{
+
+  // Asserts and checks
+  PRECICE_TRACE(dataID, size);
+  PRECICE_CHECK(_state != State::Finalized, "writeBlockVectorGradientData(...) cannot be called after finalize().");
+  PRECICE_REQUIRE_DATA_WRITE(dataID);
+  if (size == 0)
+    return;
+  PRECICE_CHECK(valueIndices != nullptr, "writeBlockVectorGradientData() was called with valueIndices == nullptr");
+  PRECICE_CHECK(values != nullptr, "writeBlockVectorGradientData() was called with values == nullptr");
+
+  // Get the data
+  WriteDataContext &context = _accessor->writeDataContext(dataID);
+  PRECICE_ASSERT(context.providedData() != nullptr);
+
+  mesh::Data &data = *context.providedData();
+
+  // Check if the Data object with ID dataID has been initialized with gradient data
+  PRECICE_CHECK(data.hasGradient(), "Data \"{}\" has no gradient values available! Please turn on the gradient attribute in the configuration file.", data.getName())
```

Comment:
```suggestion
  PRECICE_CHECK(data.hasGradient(), "Data \"{}\" has no gradient values available! Please turn on the gradient attribute in the configuration file.", data.getName())
```
```suggestion
  PRECICE_CHECK(data.hasGradient(), "Data \"{}\" has no gradient values available. Please turn on the gradient attribute in the configuration file.", data.getName())
```
Same as above. Also, could you describe how to do that "turn on the gradient attribue"?